src/sys/unix/kqueue.rs

```diff
@@ -238,6 +238,78 @@ impl Selector {
             Ok(())
         }
     }
+
+    // Used by `Awakener`.
+    #[cfg(any(target_os = "freebsd", target_os = "ios", target_os = "macos"))]
+    pub fn setup_awakener(&self, token: Token) -> io::Result<()> {
+        // First attempt to accept user space notifications.
+        let mut kevent = kevent!(
+            0,
+            libc::EVFILT_USER,
+            libc::EV_ADD | libc::EV_CLEAR | libc::EV_RECEIPT,
+            token.0
+        );
+
+        cvt(unsafe {
+            libc::kevent(
+                self.kq,
+                &kevent as *const libc::kevent,
+                1,
+                &mut kevent as *mut libc::kevent,
+                1,
+                ptr::null(),
+            )
+        })
+        .map(|_| ())?;
+
+        if (kevent.flags & libc::EV_ERROR) != 0 && kevent.data != 0 {
+            Err(io::Error::from_raw_os_error(kevent.data as i32))
+        } else {
+            Ok(())
+        }
+    }
+
+    // Used by `Awakener`.
+    #[cfg(any(target_os = "freebsd", target_os = "ios", target_os = "macos"))]
+    pub fn try_clone(&self) -> io::Result<Selector> {
```

Comment:
Since this is kqueue specific, what if we don't return a `Selector`? That would prevent needing incrementing `NEXT_ID`. We could just return the RawFd, or some other private type.

Reply:
We can also just use `0`/max as id since it won't be used in the `Awakener` anyway.

Reply:
Reusing the id seems to work fine, so I'll do that.